simulator/func_sim/alu.h

```diff
@@ -203,7 +203,17 @@ struct ALU
     // Leading zero/ones
     template<typename I, typename T> static void clo( I* instr)  { instr->v_dst = count_leading_ones<T>( instr->v_src1); }
     template<typename I, typename T> static void clz( I* instr)  { instr->v_dst = count_leading_zeroes<T>( instr->v_src1); }
-
+    template<typename I, typename T> static
+    void pcnt(I* instr){
+      std::size_t max = bitwidth<T> - 1 - count_leading_zeroes<T>( ~instr->v_src1);
+      uint8 count = 0;
+      for ( std::size_t index = 0; index < max; index++)
+      {
+        count += (instr->v_src1 >> index) & 1;
```

Comment:
You need a `narrow_cast<uint8>( ... )` here to supress warnings regarding data narrowing.
Or, alternatively, you can use ` if/?: ` operator here:
```c++
if ( ( instr->v_src1 >> index & 1) != 0)
    ++count;
```